### PR TITLE
DDev followup: use `drupal:tests` in CI for functional JS tests and reconfigure/document local testing.

### LINKED ISSUE #4538: `blt tests:drupal` broken after switch to DDev
# Problem
When I attempted to run `blt tests:drupal` after #4458 was merged in, I encountered this error:
```
[error]  Could not find Selenium. Install it via `composer require se/selenium-server-standalone`.
```
I ran the `composer require se/selenium-server-standalone` command and after that and got the following error:
```
[error]  Timed out.
...
Selenium internal logs:
```

# Steps to reproduce
See problem description

# Proposed solution
Add new testing dependency and figure out why it is timing out.

### PR DESCRIPTION
**Pull request by richardbporter**
Related to #4538 

This PR removes the BLT configuration required for the ddev chromedriver container to work from our `blt.yml` file. The reason for this is so we don't have to then override those values to the defaults provided by BLT in `ci.blt.yml`. The other benefit is should those values change in BLT, we'll get those updated values.

This change means it is necessary to specify more values in your `blt/local.blt.yml` files if you'd like to run Drupal tests locally. I've updated `tests/README.md` to reflect this.

Additionally, this PR changes how our tests our run in CI. We now use the `tests:drupal` command to run all phpunit tests and configure phpunit to use our repo config. This ensures the necessary chrome and chromedriver requirements for functional JS test.

### Random
Added `ddev phpcs`, `ddev phpcbf` and `ddev phpunit` commands because I got sick of typing `./vendor/bin/...`.

### Testing
- Reconfigure your `blt/local.blt.yml` file according to the `tests/README.md` instructions to run Drupal core/contrib test.
- Run `blt tests:drupal` and make sure it works.
- Run `blt tests:phpunit` and make sure the uiowa_search functional JS test located at https://github.com/uiowa/uiowa/blob/main/docroot/modules/custom/uiowa_search/tests/src/FunctionalJavascript/SearchTest.php is not marked as skipped.
- Make sure the same test is not marked as skipped in CI log output. 
  - Previous example of skipped test: https://app.travis-ci.com/github/uiowa/uiowa/builds/243121493#L2168
  - Latest build (could change): https://app.travis-ci.com/github/uiowa/uiowa/builds/243128919#L2171

### CODE REVIEW

**Comment by richardbporter** [comment]
Still can't get these to run https://app.travis-ci.com/github/uiowa/uiowa/builds/243070864#L2169

**Comment by pyrello** [comment]
One other thing I am noticing when trying to run a test while debugging is that it is necessary to the green play icon quite a few more times as it seems to be waiting for a prompt with every separate command that is being triggered (`blt`, `drush`, `phpunit`). This certainly isn't a deal-breaker, but slows down the process of testing quite a bit. For context, I have turned xdebug on with `ddev xdebug on` and am running `./vendor/bin/blt tests:drupal` from inside the container.

**Comment by richardbporter** [comment]
> One other thing I am noticing when trying to run a test while debugging is that it is necessary to the green play icon quite a few more times as it seems to be waiting for a prompt with every separate command that is being triggered (`blt`, `drush`, `phpunit`). This certainly isn't a deal-breaker, but slows down the process of testing quite a bit. For context, I have turned xdebug on with `ddev xdebug on` and am running `./vendor/bin/blt tests:drupal` from inside the container.

Thats unfortunate. I hadn't tested xdebug with drupal tests specifically. I don't know what would be different with ddev compared to DrupalVM for that to suddenly happen.

I tried this out and had to accept an incoming connection from the web container but it stopped in my test. It was very slow but I didn't have to continue execution three times.

File: blt/blt.yml [hunk 1]
```diff
@@ -26,12 +26,6 @@ drush:
 tests:
   phpunit:
     - config: '${repo.root}'
-  drupal:
-    web-driver: false
-    sudo-run-tests: false
-    mink-driver-args-webdriver: ''
-    simpletest-db: ''
-    simpletest-base-url: ''
```
Comment (richardbporter) [inline blt/blt.yml:34]: By removing this, we will use the default settings in `vendor/acquia/blt/config/build.yml` and then we override for our local environment. Just an FYI.

**Comment by pyrello** [comment]
Hmm... just to clarify, previously I was running `blt tests:drupal`. When I run `./vendor/bin/blt tests:drupal`, it doesn't seem to break at all.

**Comment by richardbporter** [comment]
I'm guessing the extra `blt` and `drush` sessions might be related to the hooks in `vendor/acquia/blt-drupal-test/src/Blt/Plugin/Commands/DrupalTestCommand.php::executeTests`. I think the command runs some validation through drush, blt, etc. You could temporarily remove the `@interactGenerateSettingsFiles through @executeInVm` lines in the docblock of that method.

**Comment by richardbporter** [comment]
> Hmm... just to clarify, previously I was running `blt tests:drupal`. When I run `./vendor/bin/blt tests:drupal`, it doesn't seem to break at all.

I can get it to break with `./vendor/bin/blt tests:drupal` and `ddev blt tests:drupal` (run from host) but both take a long time.

My `local.blt.yml` file that I have for this PR, contains
```
# Test config for ddev.
tests:
  drupal:
    web-driver: false
    sudo-run-tests: false
    mink-driver-args-webdriver: ''
    simpletest-db: ''
    simpletest-base-url: ''
    phpunit:
      -
        # The directory to scan for tests. Change to what you want to test.
        directory: '${docroot}/modules/contrib/entity_usage'
        # Use the core PHPUnit config file.
        config: '${docroot}/core/phpunit.xml.dist'
        # Filter by a specific test type, class, name, etc.
        filter: testEntityEmbed
```

**Comment by cory-skeers** [comment]
Less comfortable reviewing the code changes themselves, but can confirm everything went smoothly in my testing of them. Also, the added `ddev` commands were nifty and a nice random addition!

I'd be comfortable adding a review, but wanted to pause in case others wanted to get eyes on this first.

**Comment by richardbporter** [comment]
> Less comfortable reviewing the code changes themselves, but can confirm everything went smoothly in my testing of them. Also, the added `ddev` commands were nifty and a nice random addition!
> 
> I'd be comfortable adding a review, but wanted to pause in case others wanted to get eyes on this first.

Thanks. Nothing should really change after this except folks having to update their `local.blt.yml` file to run Drupal tests locally. We just kinda reversed the BLT config to make that happen. FYI @pyrello and @joewhitsitt as I know they were running Drupal tests locally.

**Review by joewhitsitt — APPROVED** [review]
reviewed, not tested

**Pull request merged**